js/app.js

```diff
@@ -11,29 +11,146 @@ Calculator.prototype.getHistoryAsString = function() {
     return this.history.join('\n');
 }
 
+// Pomocnicza funkcja walidacji (zwraca Number albo null)
+Calculator.prototype._toNumberOrNull = function(value) {
+    const n = Number(value);
+    return Number.isFinite(n) ? n : null;
```

Comment:
Czy jesteś pewny, że to działa poprawnie? `Number(value)` zwraca liczbę lub `NaN`